jaxrs-api/src/main/java/jakarta/ws/rs/core/GenericType.java

```diff
@@ -198,7 +198,7 @@ private static Class getArrayClass(final Class c) {
      */
     static Type getTypeArgument(final Class<?> clazz, final Class<?> baseClass) {
         // collect superclasses
-        Stack<Type> superclasses = new Stack<Type>();
+        ArrayDeque<Type> superclasses = new ArrayDeque<>();
```

Comment:
Why forcing array-based deque implementation on the *left* side, instead of simply using an interface as before, or using `var`?

Reply:
>instead of simply using an interface as before

Before it was using a Class - `java.util.Stack`. It's not an interface.

Reply:
You're right, I missed that.